### PR TITLE
Homewrok w10

### PR DESCRIPTION
**Pull request by MarkKolenbrander**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -58,6 +62,8 @@ dependencies {
 
     //Glide
     implementation 'com.github.bumptech.glide:glide:4.13.2'
+    testImplementation 'junit:junit:4.13.2'
+    testImplementation 'junit:junit:4.13.2'
```
Comment (zlippard) [inline app/build.gradle:66]: Looks like this line is duplicated

File: app/build.gradle [hunk 2]
```diff
@@ -97,4 +108,11 @@ dependencies {
     testImplementation 'org.threeten:threetenbp:1.4.0'
     testImplementation 'com.squareup.retrofit2:retrofit-mock:2.6.0'
     testImplementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.4"
+
+    //Hilt Testing
+    androidTestImplementation "com.google.dagger:hilt-android-testing:2.44"
+    kaptAndroidTest "com.google.dagger:hilt-android-compiler:2.44."
```
Comment (zlippard) [inline app/build.gradle:114]: looks like there's an extra dot (.) at the end here:

```suggestion
    kaptAndroidTest "com.google.dagger:hilt-android-compiler:2.44"
```

File: app/src/main/java/com/markkolenbrander/capstonenewsapp/database/converters/DataConverter.kt [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
 import androidx.room.TypeConverter
+import com.google.gson.Gson
 import com.google.gson.reflect.TypeToken
-import com.markkolenbrander.capstonenewsapp.App
 import com.markkolenbrander.capstonenewsapp.models.Source
 
-
 class DataConverter {
 
+    private val gson = Gson()
```
Comment (zlippard) [inline app/src/main/java/com/markkolenbrander/capstonenewsapp/database/converters/DataConverter.kt:10]: I know we were trying to figure out how to do this with Hilt and inject a `Gson` instance.  From what I've seen online with other examples this seems like the best/simplest approach. Nice work!

File: app/src/main/java/com/markkolenbrander/capstonenewsapp/views/NewsArticleDatabase.kt [hunk 4]
```diff
@@ -18,19 +16,19 @@ private const val DATABASE_VERSION = 1
...
+//        fun buildDatabase(context: Context): NewsArticleDatabase {
+//            return Room.databaseBuilder(
+//                context,
+//                NewsArticleDatabase::class.java,
+//                DATABASE_NAME,
+//            )
+//                .allowMainThreadQueries()
+//                .build()
+//        }
+//    }
```
Comment (zlippard) [inline app/src/main/java/com/markkolenbrander/capstonenewsapp/views/NewsArticleDatabase.kt:31]: Minor, but you could leave this here and reference it via your AppModule with `NewsArticleService.buildDatabase(...)`

**Review by zlippard — APPROVED** [review]

**Comment by MarkKolenbrander** [comment]
Thanks Zac!

**Pull request merged**